e2e/vm/soci_test.go

```diff
@@ -79,6 +85,24 @@ var testSoci = func(o *option.Option, installed bool) {
 			command.Run(o, "rmi", "-f", ffmpegSociImage)
 			gomega.Expect(finchPullMounts).Should(gomega.Equal(nerdctlPullMounts))
 		})
+		ginkgo.It("finch push should work", func() {
+			resetVM(o, installed)
+			resetDisks(o, installed)
+			writeFile(finchConfigFilePath, []byte("cpus: 6\nmemory: 4GiB\nsnapshotters:\n    "+
+				"- soci\nvmType: qemu\nrosetta: false"))
+			command.New(o, virtualMachineRootCmd, "init").WithTimeoutInSeconds(600).Run()
+			port = fnet.GetFreePort()
+			command.New(o, "run", "-dp", fmt.Sprintf("%d:5000", port), "--name", "registry", registryImage).
+				WithTimeoutInSeconds(30).Run()
+			buildContext := ffs.CreateBuildContext(fmt.Sprintf(`FROM %s
+		CMD ["echo", "bar"]
+			`, ubuntuImage))
+			ginkgo.DeferCleanup(os.RemoveAll, buildContext)
+			targetTag := fmt.Sprintf(`localhost:%d/test-push-soci:tag`, port)
+			command.New(o, "build", "-t", targetTag, buildContext).WithTimeoutInSeconds(30).Run()
```

Comment:
Can we verify ztocs and SOCI index are created by shelling into VM and checking `soci info` and/or `soci index list` ? https://github.com/awslabs/soci-snapshotter/blob/main/docs/getting-started.md#optional-inspect-soci-index-and-ztoc